make_github_action.py

```diff
@@ -0,0 +1,156 @@
+        else: # pick something random
+            counter = 0 # prevent infinite loop
+            while True:
+                counter += 1
+                newtag = 'tag_' + str(random.randint(100000,999999))
+                if newtag not in tag_vars.values():
+                    tag_vars[tag] = newtag
+                    return newtag
+                assert counter < 1000
+
+# output start of the file checking job
+def account_job_check(username: str, infos):
+    ret = f'''\
+  # this job checks which containers for account "{username}" should be updated
+  check-files-{username}:
+    name: Check modified files ({username})
+    runs-on: ubuntu-latest
+    # variables that tell which containers should be updated
+    outputs:
+'''
+    for name,tag,dir in infos:
+        tag = tag_modifier(tag)
+        ret += f'''\
+      {name}__{tag}: ${{{{ steps.check-files-{username}.outputs.{name}__{tag} }}}}
+'''
+    ret += f'''\
+    steps:
+    - name: Code checkout
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 2
+    - name: Look for modified files
+      id: check-files-{username}
+      # look for whether any files were modified in the container's directory
+      run: |
+'''
+    for name,tag,dir in infos:
+        tag = tag_modifier(tag)
+        ret += f'''\
+        COUNT=$(git diff --name-only HEAD^ HEAD | grep {dockerhub_accounts[username]}/{name}/ | wc -l)
+        if [[ $COUNT != 0 ]]; then echo "{name}__{tag}=true" >> $GITHUB_OUTPUT; else echo "{name}__{tag}=false" >> $GITHUB_OUTPUT; fi;
+'''
+    return ret
+
+# output the start of a per-account job
+def account_job_start(username: str):
+    return f'''\
+  # this job pushes container images for account "{username}"
+  push-images-{username}:
+    name: Push docker images ({username})
+    runs-on: ubuntu-latest
+    needs: check-files-{username}
+    steps:
+    - name: Check out repo
+      uses: actions/checkout@v3
+    - name: Docker Hub login
+      uses: docker/login-action@releases/v1
+      with:
+        username: ${{{{ secrets.DOCKER_USERNAME_{username.upper()} }}}}
+        password: ${{{{ secrets.DOCKER_PASSWORD_{username.upper()} }}}}
+'''
+
+# output a job step for pushing a container
+def account_container_string(username: str, name: str, tag: str, path: str):
+    return f'''\
+    - name: Build and push {username}/{name}:{tag}
+      if: needs.check-files-{username}.outputs.{name}__{tag_modifier(tag)} == 'true'
+      uses: docker/build-push-action@releases/v3
+      with:
+        context: {path}
+        push: true
+        tags: {username}/{name}:{tag}
+'''
+
+# find all metadata.json files and create a job step for each
+def process_account(username: str, filepath: str):
+    log_err(f'processing docker hub account {username}')
+    infos = [] # list of (name,tag,dir)
+    # traverse the directory tree in sorted order to keep diff sizes small
+    for dir,subdirs,files in sorted(os.walk(filepath)):
+        if 'metadata.json' in files:
```

Comment:
Do we just want to read dockerfiles which have metadata.json? Is this a check of some sort?